### DISCUSSION TITLE
maven repository not being looked up

### DISCUSSION THREAD

**Discussion by beachygreg** [renovatebot/renovate#19537] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.65

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab (.com and CE/EE) 15.0.5-ee

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

When we upgraded renovate bot from 34.35 to 34.65 the lookup of our gradle projects to our authenticated maven repositories stopped working. 

Our projects are gradle projects that reference gitlab maven repositories, the repositories are specified in host rules with gitlab tokens that allow access to those repositories. I did not see any errors in the logs for access denied so the issue could be to do with the discovery of the maven repositories in the gradle files, I also see the hosts config in the logs configured correctly. 

### Relevant debug logs

<details><summary>Logs </summary>

<description>The host rule looks to be applied:</description>
```
{"name":"renovate","hostname":"xxxx","pid":14,"level":20,"logContext":"87g29lt2b3Zg8sVWFtCtu","msg":"Adding token authentication for https://<our gitlab server>/api/v4/projects/<gitlab project id>/packages/maven to hostRules","time":"2022-12-20T07:16:17.232Z","v":0}
```

</details>

<details><summary>Logs</summary>
<description>There should be a lookup to our internal repo but it only looks in public repos</description>

```
{"name":"renovate","hostname":"xxxx","pid":14,"level":20,"logContext":"87g29lt2b3Zg8sVWFtCtu","repository":"group_name/examples/demo-app","msg":"Looking up com.company.name:plugin in repository https://repo.maven.apache.org/maven2/","time":"2022-12-20T07:16:19.309Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Without logs or a reproduction, we would need you to identify exactly which release it went from working to non working

**Comment by beachygreg** [comment]
I knows its between 34.35 and 34.50 something so I will run a few jobs and try to narrow it down

**Comment by beachygreg** [comment]
Okay so I am pretty confused. 
When I run with the docker `image: renovate/renovate:34.25.1@sha256:0c7a1e3c9ab39a31ddeac6eb37db896491bcb21f3b4a4a64e4fcf02f2d491271` everything works but if I run with `renovate/renovate:34.25` it does not work. 
If I look at the logs the version of renovate is the same. If the docker image got overwritten for that tag it might explain something but otherwise I am at a loss.

I would assume our gitlab instance is caching the image, are the releases immutable for the docker tags?

**Comment by rarkins** [comment]
Tags aren't immutable, SHAs are. You will see X.Y tags change but X.Y.Z shouldn't often

**Comment by beachygreg** [comment, marked as answer]
Sorry this does not seem to be a bug, I have had the error and not in the same version docker image so its something environmental. I suspect that the requests are getting rate limited, but I don't see anything in the log. The maven registries are listed in hosts correctly but once the lookups stop working they are just not in the registries list for the dependency and thus the lookup does not work.

**Comment by beachygreg** [comment]
Is there any validation on host authentication for registries in the hosts file? and if there is I think it does not log failures correct?

**Reply by rarkins** [reply]
You should see errors logged at debug level. e.g. run with LOG_LEVEL=debug in env

**Reply by beachygreg** [reply]
I set the level to debug, but I don't see anything in the log. It was also the case before I figured out how to add the authentication. 
The registry is getting picked up from gradle fine, but when the authentication is failing then it seems to just ignore the maven registries.

**Reply by beachygreg** [reply]
It is possible I am missing the message (its a very large log file), what should I be looking for to spot maven registry authentication failure?

**Reply by rarkins** [reply]
You can look under the source code for lib/modules/datasource/maven to see what type of debug messages to expect
